src/marqo/core/index_management/index_management.py

```diff
@@ -75,35 +47,66 @@ def __init__(self, vespa_client: VespaClient, zookeeper_client: Optional[Zookeep
                 the object can create/delete indexes, otherwise, it raises an InternalError during index operations.
         """
         self.vespa_client = vespa_client
-        self._zookeeper_client = zookeeper_client
-        self._zookeeper_deployment_lock: Optional[ZookeeperDistributedLock] = self._instantiate_deployment_lock()
+        self._zookeeper_deployment_lock = get_deployment_lock(zookeeper_client, deployment_lock_timeout) \
+            if zookeeper_client else None
         self._enable_index_operations = enable_index_operations
+        self._deployment_timeout_seconds = deployment_timeout_seconds
+        self._convergence_timeout_seconds = convergence_timeout_seconds
 
-    def _instantiate_deployment_lock(self) -> Optional[ZookeeperDistributedLock]:
-        """Instantiate a ZookeeperDistributedLock."""
-        if self._zookeeper_client is None:
-            return None
-        else:
-            return get_deployment_lock(self._zookeeper_client)
+    @staticmethod
+    def validate_index_settings(index_name: str, settings_dict: dict) -> None:
+        """
+        Validates index settings using the IndexSettings model.
+
+        Args:
+            index_name (str): The name of the index to validate settings for.
+            settings_dict (dict): A dictionary of settings to validate.
+
+        Raises:
+            ValidationError: If the settings are invalid in the context of the IndexSettings model.
+            api_exceptions.InvalidArgError: If several settings are invalid in the context of the Marqo API.
+            Check the errors in to_marqo_index_request method of IndexSettings model for more details.
+
+        Returns:
+            None: If the validation is successful, nothing is returned,
+            else InvalidArgumentError is raised.
+        """
+        index_settings = IndexSettings(**settings_dict)
+        index_settings.to_marqo_index_request(index_name)
 
     def bootstrap_vespa(self) -> bool:
         """
         Add Marqo configuration to Vespa application package if an existing Marqo configuration is not detected.
 
         Returns:
-            True if Vespa was configured, False if it was already configured
+            True if Vespa was bootstrapped, False if it was already up-to-date
         """
-        app = self.vespa_client.download_application()
-        configured = self._marqo_config_exists(app)
+        with self._vespa_deployment_lock():
+            application = self._get_application_context_manager_for_bootstrapping_and_rollback(check_configured=False)
+            with application as app:
+
+                marqo_version = version.get_version()
+                has_marqo_settings_schema = app.has_schema(self._MARQO_SETTINGS_SCHEMA_NAME)
+                marqo_config_doc = self._get_marqo_config() if has_marqo_settings_schema else None
+
+                if not app.need_bootstrapping(marqo_version, marqo_config_doc):
```

Comment:
Add some debug messages to this process. They've proven very useful in the past

Reply:
How will we use them? Can we enable debug logging in Cloud dynamically?